Source/CarthageKitTests/Fakes/FakeURLSession.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+
+final class FakeDownloadTask: NSURLSessionDownloadTask {
+	var started = false
+	override func resume() {
+		self.started = true
+	}
+
+	var cancelled = false
+	override func cancel() {
+		self.cancelled = true
+	}
+}
+
+final class FakeDataTask: NSURLSessionDataTask {
+	var started = false
+	override func resume() {
+		self.started = true
+	}
+
+	var cancelled = false
+	override func cancel() {
+		self.cancelled = true
+	}
+}
+
+final class FakeURLSession: NSURLSession {
```

Comment:
If I were to mock out the entire url loading mechanism, I'd use my own (MockHTTP)[https://github.com/younata/MockHTTP], but I'd much rather not have to go down to that level - it's entirely unnecessary (would easily add 10x more test setup - would you like to do that the next time we add another network service for Carthage to talk to?), and providing a fake url session is much cleaner because we're forcing the issue of needing to have a network interface injected by the caller, not leaving it ambiguous as to who's job that is.